src/components/MTableEditCell/index.js

```diff
@@ -72,61 +56,48 @@ function MTableEditCell(props) {
   };
 
   const onApprove = () => {
-    setState({ ...state, isLoading: true });
+    setState({ ...state, isLoading: true }, () => {
+      props.cellEditable
+        .onCellEditApproved(
+          state.value, // newValue
+          props.rowData[props.columnDef.field], // oldValue
+          props.rowData, // rowData with old value
+          props.columnDef // columnDef
+        )
+        .then(() => {
+          setState({ ...state, isLoading: false });
+          props.onCellEditFinished(props.rowData, props.columnDef);
+        })
+        .catch((error) => {
+          // might be wrong
+          setState({ ...state, isLoading: false, error });
+        });
+    });
   };
 
   const onCancel = () => {
     props.onCellEditFinished(props.rowData, props.columnDef);
   };
 
-  function renderActions() {
-    if (state.isLoading) {
-      return (
-        <div style={{ display: 'flex', justifyContent: 'center', width: 60 }}>
-          <CircularProgress size={20} />
-        </div>
-      );
-    }
-
-    const actions = [
-      {
-        icon: props.icons.Check,
-        tooltip: props.localization && props.localization.saveTooltip,
-        onClick: onApprove,
-        disabled: state.isLoading
-      },
-      {
-        icon: props.icons.Clear,
-        tooltip: props.localization && props.localization.cancelTooltip,
-        onClick: onCancel,
-        disabled: state.isLoading
-      }
-    ];
-
-    return (
-      <props.components.Actions
-        actions={actions}
-        components={props.components}
-        size="small"
-      />
-    );
-  }
-
   return (
     <TableCell size={props.size} style={getStyle()} padding="none">
       <div style={{ display: 'flex', alignItems: 'center' }}>
         <div style={{ flex: 1, marginRight: 4 }}>
           <props.components.EditField
             columnDef={props.columnDef}
             value={state.value}
-            onChange={(prevState, value) => setState({ ...prevState, value })}
+            onChange={(value) => setState({ value })}
```

Comment:
This spreading is also needed since you would override the loading state, This works for class but not for function components.